commands.go

```diff
@@ -788,6 +788,50 @@ func (c cmdable) Set(ctx context.Context, key string, value interface{}, expirat
 	return cmd
 }
 
+// SetArgs provides arguments for the SetWithArgs command.
+//
+// Mode can be `NX` or `XX` or empty.
+//
+// Zero expiration means the key has no expiration time.
+// KeepTTL(-1) expiration is a Redis KEEPTTL option to keep existing TTL.
+//
+// When Get is true, the command returns the old value stored at key, or nil when key did not exist.
+type SetArgs struct {
+	Mode     string
+	ExpireAt time.Duration
```

Comment:
I believe ExpireAt should be called TTL.

And there is no support for `EXAT timestamp|PXAT milliseconds-timestamp`. II is fine if you decide not to support it in this PR, but I believe we need:

```go
TTL time.Duration // for px/ex
ExpireAt time.Time // for pxat/exat
```